plumbing/transport/http/upload_pack.go

```diff
@@ -171,10 +171,13 @@ func uploadPackRequestToReader(req *packp.UploadPackRequest) (*bytes.Buffer, err
 		return nil, fmt.Errorf("sending upload-req message: %s", err)
 	}
 
-	if err := req.UploadHaves.Encode(buf); err != nil {
+	if err := req.UploadHaves.Encode(buf, false); err != nil {
```

Comment:
Here UploadRequest is adding a flush, but UploadHaves is not? This seems a partial fix, since it's still against the HTTP protocol as specified here: https://github.com/git/git/blob/4e59582ff70d299f5a88449891e78d15b4b3fabe/Documentation/technical/http-protocol.txt#L302

Let's be triple check this.